lib/exercises.js

```diff
@@ -1,17 +1,85 @@
 // This method will return an array of arrays.
 // Each subarray will have strings which are anagrams of each other
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: O(n^3) - outer while loop (line 10) has a nested for loop (lines 24 and 27), it also has .splice() nested in a for loop, but this should still be O(n^3) overall.
+// Space Complexity: O(n) - result, hash, and matched will be at most length n, where n is the number of elements in the strings array. Even though result is a nested array, it will never contain more than n elements.
 function grouped_anagrams(strings) {
```

Comment:
Suggestion:  Sort the letters in each word and use that as a key to a hash and make the value a list of words with those letters.